internal/server/redis/redis.go

```diff
@@ -0,0 +1,37 @@
+package redis
+
+import (
+	"fmt"
+
+	"github.com/go-redis/redis/v8"
+)
+
+type Redis struct {
+	client *redis.Client
+}
+
+type Options struct {
+	Host     string
+	Port     int
+	Username string
+	Password string
+	Options  string
+}
+
+func NewRedis(options Options) (*Redis, error) {
+	var client *redis.Client
+
+	if len(options.Host) > 0 {
+		redisOptions, err := redis.ParseURL(fmt.Sprintf("redis://%s:%d?%s", options.Host, options.Port, options.Options))
+		if err != nil {
+			return nil, fmt.Errorf("invalid redis options: %v", options)
+		}
+
+		redisOptions.Username = options.Username
+		redisOptions.Password = options.Password
+
+		client = redis.NewClient(redisOptions)
```

Comment:
Do you need to use the cluster client here? I'm assuming we would want some kind of redundancy.

Reply:
It depends how the Redis cluster is configured. Managed services like GCP's MemoryStore and AWS's ElastiCache can implement redundancy without resorting to Redis Clusters. Implementing other types of redundancy will require more complex configurations such as sentinels or cluster configurations.

For the time being, we can get away with implementing just the basic configuration